ports/zephyr/zephyr_storage.c

```diff
@@ -244,9 +258,7 @@ static const mp_rom_map_elem_t zephyr_flash_area_locals_dict_table[] = {
     { MP_ROM_QSTR(MP_QSTR_readblocks), MP_ROM_PTR(&zephyr_flash_area_readblocks_obj) },
     { MP_ROM_QSTR(MP_QSTR_writeblocks), MP_ROM_PTR(&zephyr_flash_area_writeblocks_obj) },
     { MP_ROM_QSTR(MP_QSTR_ioctl), MP_ROM_PTR(&zephyr_flash_area_ioctl_obj) },
-    #if FIXED_PARTITION_EXISTS(storage_partition)
-    { MP_ROM_QSTR(MP_QSTR_STORAGE), MP_ROM_INT(FIXED_PARTITION_ID(storage_partition)) },
-    #endif
+    DT_FOREACH_STATUS_OKAY(fixed_partitions, FOREACH_PARTITION)
```

Comment:
Please put a high-level comment about what all this is doing, for future readers.

(Don't have to explain exactly, just something like `// Generate named lookup of all partition IDs from DT flash area definitions`)

Reply:
Done